core/src/main/java/tc/oc/pgm/core/CoreModule.java

```diff
@@ -71,6 +71,20 @@ public MatchModule createMatchModule(Match match) {
       return ImmutableList.of(RegionModule.class, TeamModule.class);
     }
 
+    public ImmutableSet<SelfIdentifyingFeatureDefinition> parseModeSet(
+        Node node, MapFactory context) throws InvalidXMLException {
```

Comment:
```suggestion
    public ImmutableSet<Mode> parseModeSet(Node node) throws InvalidXMLException {
```
Make the factory a field in the class, no need to pass it as parameter